client/src/components/ReviewsList/index.jsx

```diff
@@ -31,24 +35,67 @@ class ReviewsList extends React.Component {
     this.getReviews();
   }
 
-  componentDidUpdate(prev) {
+  componentDidUpdate(prev, previousState) {
     const { currentProduct } = this.props;
+    const { sortList } = this.state;
     const { currentProduct: previousProduct } = prev;
+    const { sortList:  priorSortList } = previousState;
     if (currentProduct !== previousProduct) {
       this.getReviews();
     }
+    if (sortList !== priorSortList) {
+      this.getReviews();
+    }
```

Comment:
Next time it's better make sure you only call the API once by declaring a variable like `shouldGetReviews` and conditionally set it to true depending on the situation. Then at the end, check if you should call the API and call it.

```suggestion
    let shouldGetReviews = false;
    
    if (currentProduct !== previousProduct) {
      shouldGetReviews = true;
    }
    if (sortList !== priorSortList) {
      shouldGetReviews = true;
    }
    
    if (shouldGetReviews === true) {
      this.getReviews();
    }
```